service/src/test/java/uk/gov/hmcts/reform/fpl/service/ManageDocumentServiceTest.java

```diff
@@ -529,6 +525,85 @@ void shouldNotUpdatePreviousSupportingEvidenceEntryWhenNoUpdatesWhereMade() {
+            .manageDocument(buildFurtherEvidenceManagementDocument(YES.getValue()))
+            .build();
+
+        CaseData caseDataBefore = CaseData.builder()
+            .hearingFurtherEvidenceDocuments(new ArrayList<>(List.of(
+                element(hearingId, HearingFurtherEvidenceBundle.builder()
+                    .supportingEvidenceBundle(previousSupportingEvidenceList)
+                    .build()))))
+            .build();
+
+        List<Element<SupportingEvidenceBundle>> updatedEvidenceBundle =
+            manageDocumentService.setDateTimeOnHearingFurtherEvidenceSupportingEvidence(caseData, caseDataBefore);
+
+        SupportingEvidenceBundle firstSupportingEvidenceBundle = updatedEvidenceBundle.get(0).getValue();
+        SupportingEvidenceBundle secondSupportingEvidenceBundle = updatedEvidenceBundle.get(1).getValue();
+
+        assertThat(updatedEvidenceBundle.size()).isEqualTo(2);
+        assertThat(firstSupportingEvidenceBundle.getDateTimeUploaded()).isEqualTo(time.now());
+        assertThat(secondSupportingEvidenceBundle.getDateTimeUploaded()).isEqualTo(time.now());
+    }
+
+    @Test
+    void shouldNotUpdatePreviousSupportingEvidenceWhenFurtherEvidenceIsAssociatedWithAHearing() {
+        List<Element<SupportingEvidenceBundle>> previousSupportingEvidenceList
+            = buildSupportingEvidenceBundle(futureDate);
+
+        UUID hearingId = UUID.randomUUID();
+        HearingBooking hearingBooking = buildFinalHearingBooking();
+
+        CaseData caseData = CaseData.builder()
+            .hearingDetails(List.of(element(hearingId, hearingBooking)))
+            .manageDocumentsHearingList(buildDynamicList(hearingId))
+            .supportingEvidenceDocumentsTemp(List.of(
+                previousSupportingEvidenceList.get(0),
+                element(SupportingEvidenceBundle.builder().build())))
+            .manageDocument(buildFurtherEvidenceManagementDocument(YES.getValue()))
+            .build();
+
+        CaseData caseDataBefore = CaseData.builder()
+            .hearingFurtherEvidenceDocuments(new ArrayList<>(List.of(
+                element(hearingId, HearingFurtherEvidenceBundle.builder()
+                    .supportingEvidenceBundle(previousSupportingEvidenceList)
+                    .build()))))
+            .build();
+
+        List<Element<SupportingEvidenceBundle>> updatedEvidenceBundle =
+            manageDocumentService.setDateTimeOnHearingFurtherEvidenceSupportingEvidence(caseData, caseDataBefore);
+
+        SupportingEvidenceBundle firstSupportingEvidenceBundle = updatedEvidenceBundle.get(0).getValue();
+        SupportingEvidenceBundle secondSupportingEvidenceBundle = updatedEvidenceBundle.get(1).getValue();
+
+        assertThat(updatedEvidenceBundle.size()).isEqualTo(2);
+        assertThat(firstSupportingEvidenceBundle).isEqualTo(previousSupportingEvidenceList.get(0).getValue());
+        assertThat(secondSupportingEvidenceBundle.getDateTimeUploaded()).isEqualTo(time.now());
```

Comment:
Could we assert a bit more? We are checking only date, is it enough?

Reply:
I think in this case it's fine. Second instance of supporting evidence document will be empty so we are just asserting that dateAndTime is appended.